pkg/costmodel/metrics.go

```diff
@@ -609,9 +631,29 @@ func (cmme *CostModelMetricsEmitter) Start() bool {
 				if len(costs.RAMAllocation) > 0 {
 					cmme.RAMAllocationRecorder.WithLabelValues(namespace, podName, containerName, nodeName, nodeName).Set(costs.RAMAllocation[0].Value)
 				}
+				if len(costs.RAMReq) > 0 && costs.RAMReq[0].Value > 0 {
+					req := costs.RAMReq[0].Value
+					usage := 0.0
+					if len(costs.RAMUsed) > 0 {
+						usage = costs.RAMUsed[0].Value
+					}
+					idle := (req - usage) / req
```

Comment:
Thanks for the PR @macb! In cases where `usage > req`, this would result in negative idle? Is this expected behavior for the metric?

Reply:
Ya, I wasn't quite sure what the desired behavior would be. Setting negative idles to 0 means we'd lose the context that a pod is consistently overstepping its requests (but presumably not meeting its limits).

In the simple case of milliCPU:
Request: 1000m
Usage: 1500m
`(1000 - 1500) / 1000` would be and idle of `-0.5`

Converting that to an efficiency %:
`1 - (-0.5)` would be 1.5 or 150% efficiency

Keeping that context seemed better than constraining the value between 0 and 1.

Reply:
I would strongly recommend using `allocated` (i.e. `max(usage, request)`) here instead of `request` for three reasons:
1. If you are summing this value over time, and you allow negative idle, then you make it possible to have a big positive idle followed by a big negative idle---which would sum to zero idle. I.e. the value zero no longer means "no idle" it just means "net zero idle" which is very different.
2. It's not uncommon for request to be 0.0, in which case you either have to divide by zero and yield `idle=NaN`, or not export a metric for a given container. Neither is particularly nice.
3. It makes sense that a workload using all of its requested resources _and more_ would have no idle. Zero idle seems more appropriate for such a case than a negative value that requires effort to interpret.